src/functionalTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/TransformationTest.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler;
+
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.services.ExceptionRecordToCaseTransformer;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.HttpStatus.UNPROCESSABLE_ENTITY;
+
+@ExtendWith(SpringExtension.class)
+@TestPropertySource("classpath:application.conf")
+public class TransformationTest {
+
+    private final TestHelper testHelper = new TestHelper();
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void should_transform_exception_record_successfully_without_any_warnings() {
+        Response response = testHelper.postWithBody(
+            "/transform-exception-record",
+            TestHelper.fileContentAsBytes("exception-records/valid.json")
+        );
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        JsonPath transformationResponse = response.getBody().jsonPath();
+
+        assertThat(transformationResponse.getList("warnings")).isEmpty();
+
+        Map<String, Object> details = transformationResponse.getMap("case_creation_details");
+
+        assertThat(details.get("case_type_id")).isEqualTo(ExceptionRecordToCaseTransformer.CASE_TYPE_ID);
+        assertThat(details.get("event_id")).isEqualTo(ExceptionRecordToCaseTransformer.EVENT_ID);
+
+        Map<String, Object> caseData = (Map<String, Object>) details.get("case_data");
+
+        assertThat(caseData.get("legacyId")).isNull();
```

Comment:
can we use soft assertions or compare object instead of having these many asserts ?

Reply:
Construction for response object from json is not provided. Not adding just for testing purposes.

Another point: we should just not assert full objects anyway since it is done in integration tests. Key point is verifying app is up and behaving as expected.

That being said, I'm happy to leave as is (full object + maybe softly asserting if it's helping in any way) or just remove the whole thing and assert anything what could raise a warning or so